src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -317,6 +317,11 @@ public Task<JObject> GetJObjectAsync(Uri uri, ILogger log, CancellationToken tok
                             await UpdateHttpClient(promptCredentials);
                             continue;
                         }
+
+                        // null means cancelled by user
+                        // block subsequent attempts to annoy user with prompts
+                        _authRetries = HttpHandlerResourceV3Provider.MaxAuthRetries;
```

Comment:
Why is _authRetries a global field of for this source, instead of a counter just for the current request?